### ISSUE TITLE
Confusing players in BA

### ISSUE THREAD

**Issue by lyra133** [quisquous/cactbot#392]
Hello,
I'm not 100% if its a cactbot issue but let me explain first what happened :)

Today I did a BA run with some premade groups. I(SMN) was in a group with a SCH and noticed that it confused me with the scholar and vice versa. To be more precise, in my dps overlay it said "YOU" at the scholar's bar and my dps bar had just my name like it was another player on it...maybe it confused us because we have some equal abilities, for instance miasma, bio, shadowflare etc...or it could be a bug in general? Unfortunately I didn't ask the scholar if it was the other way around on his end.

But as soon we started with Ozma more weird things happened. Cactbot didn't give me any alerts when I had the meteor, THEN when the scholar got it, it told me that I have it........

I used to trust act and cactbot alot but this was mindblowing. I'm glad that I did react fast enough, otherwise i would've died for no reason. 

Is there anything I can do to fix that, maybe some wrong settings etc.? If not, I hope that you can take a look at it. Otherwise I'll need to fall back to regular act custom triggers(witch are slower and not that accurate anyway).

**Comment by quisquous** [comment]
Hmm!  That's very curious.  Do you still have the Network log file?  Can you zip it up and attach it here if you can?  Were you streaming?

**Comment by lyra133** [comment]
Hi,
not I didn't stream it. But I got the logfile. Its quite big because I did O12 weekly before the BA run. BA starts at line 228765..i think :) By the way, my name was "Lyra Venomous" and the scholars "Kioki Kun" if that helps.
[Network_20190321.2019.03.22.zip](https://github.com/quisquous/cactbot/files/2996981/Network_20190321.2019.03.22.zip)

**Comment by DrLippe** [comment]
For some reason the scholars name was got recognized as your name. From the log file i'd say its a bug related to the plugin not cactbot. The overlays got wrong data and did behave accordingly. 😱

**Comment by lyra133** [comment]
That's what i've thought. Can you confirm this, quisquous? I'll open an issue in ravahns github if you 100% do confirm.

**Comment by quisquous** [comment]
One thing that's curious is that importing the log file makes it look as though you are a SCH, even though from the log there is no point that you cast any SCH abilities.

![image](https://user-images.githubusercontent.com/462317/54884072-cce9a280-4e29-11e9-8fbf-b8b6160ed009.png)

**Comment by quisquous** [comment]
Re: meteors, it doesn't look like you ever got a meteor, at least from the logs:

```
[16:02:09.000] 00:0839:The Proto Ozma Containment Unit will be sealed off in 15 seconds!
[16:02:24.000] 00:0839:The Proto Ozma Containment Unit is sealed off!

[16:04:27.968] 1B:1041B634:Beoth Ravalris:0000:0000:0039:0000:0000:0000:
[16:04:27.968] 1B:107017FF:Elana Elhaka:0000:0000:0039:0000:0000:0000:
[16:04:27.968] 1B:107164D1:Leo Felidae:0000:0000:0039:0000:0000:0000:
[16:04:27.968] 1B:106AF205:Kurasshu Kyodaina:0000:0000:0039:0000:0000:0000:
[16:04:27.968] 1B:1070AB9B:Nanaki Mil'mania:0000:0000:0039:0000:0000:0000:
[16:04:27.973] 1B:106AA0FF:Mikoto Tribal:0000:0000:0039:0000:0000:0000:

[16:07:53.203] 1B:10250A7D:Yoyorito Buburito:0000:0000:0039:0000:0000:0000:
[16:07:53.203] 1B:107B2EAF:Tonk Hedidit:0000:0000:0039:0000:0000:0000:
[16:07:53.203] 1B:1074D0F3:Shalyssa Vanih:0000:0000:0039:0000:0000:0000:
[16:07:53.203] 1B:106A7340:Rodi Ormesang:0000:0000:0039:0000:0000:0000:
[16:07:53.203] 1B:10662975:Lilith Weaver:0000:0000:0039:0000:0000:0000:
[16:07:53.203] 1B:106EDC4A:Tataru Taru:0000:0000:0039:0000:0000:0000:

[16:09:32.707] 1B:1067E19B:Kioki Kun:0000:C2D3:0039:0000:0000:0000:
[16:09:32.707] 1B:10662975:Lilith Weaver:0000:C2D3:0039:0000:0000:0000:
[16:09:32.707] 1B:107A6D1D:Sir Fahad:0000:C2D3:0039:0000:0000:0000:
[16:09:32.707] 1B:106B0559:Hugo Fact:0000:C2D3:0039:0000:0000:0000:
[16:09:32.707] 1B:1071DA92:Xefa Badyucho:0000:C2D3:0039:0000:0000:0000:
[16:09:32.707] 1B:107B6090:Fungwan Leviathan:0000:C2D3:0039:0000:0000:0000:

[16:12:52.000] 00:0839:The Proto Ozma Containment Unit is no longer sealed!
```

Those ids also look correct:
```
:1076A0AD:Lyra Venomous:
:1067E19B:Kioki Kun:
```

I uploaded your hydatos logs so I could use the fflogs replay feature and see where people were moving around: https://www.fflogs.com/reports/3MDTxftyL7b8md6G/#fight=46&view=replay

I can definitely see you moving out for the meteors at 7m20s that Kioki has and then moving back in.  However, I don't see you move out for meteors at 2m30s or 5m50s either.  Which meteor did you have?

I am not 100% sure if this is an ACT issue or a cactbot issue.  Regarding the plugin type, do you use cactbot as the plugin type for your dps overlay, or are you using overlay plugin directly or something else?  For what it's worth, this is the only time I've ever heard of this happening.  It might be worth filing with ravahn as well, but I am not sure what else to do at this point.

**Comment by lyra133** [comment]
Thank you for digging through that long logfile.
Actually I don't remember anymore witch meteor it was. I did like 15 BA runs in total and like 8 after that one, sorry :' It might be the very first one. 
Maybe it wasnt the meteor, but the other debuff, the bomb where you have to stop attacking and moving. 

For plugin type I use cactbot. That is why I contacted you first ^^

I understand that its not a common issue- happend to me only once too, but isn't there a way to prevent that from happening in future? For now, i guess, all I can do is restart ACT, or clear all data and reload the overlays if something like this happens again.

Can you contact ravahn about this, since you have more knowledge about the plugins and can point him to specific lines in the log. Also I think you can explain it much better then I ever could :)

**Comment by DrLippe** [comment]
> One thing that's curious is that importing the log file makes it look as though you are a SCH, even though from the log there is no point that you cast any SCH abilities.
> 
> ![image](https://user-images.githubusercontent.com/462317/54884072-cce9a280-4e29-11e9-8fbf-b8b6160ed009.png)
> 
> I'm going to look into this a little bit further today and will report back here.

From this point everything what followed did behave according to ACT data. Cactbot received this faulty data.
`data.me = scholarName` so trigger fires when sombody else got the condition. 
In emulator both players are recognized correct and so are the fired triggers.

**Comment by quisquous** [comment]
Thanks for the followup, re: meteor vs acceleration bomb.  That makes a lot of sense.  The log confirms that as well.

Given that I've never seen this before, I don't know that there's anything you can do to avoid it.  Please keep me posted if you ever see it again.  I'm hopeful that the import of the log might lead to something ravahn can track down.  Sorry that I don't have any better answers for you here!

**Comment by quisquous** [comment]
I do think if you notice it happening, that restarting ACT is probably your best bet.

**Comment by ravahn** [comment]
Sorry for the delay, but I did find some information as to the reason ACT got confused.  When changing zones, ACT logs the identity of the parsing player in order to determine who 'you' are.  It finds this identity by looking at the array of combatants and picking the first one.  I've never seen this before, but for some reason it is picking the wrong identity - here's the log line:
`02|2019-03-21T22:08:19.3710000+01:00|1067e19b|Kioki Kun|ec7e568fb981099f19462bbbcb733564`

Everything else is likely a cascading result of this problem - if ACT doesn't know who 'you' are accurately, triggers probably wont work, and the overlays wont show the right information.  However, since this is the first time I've encountered this problem, it seems to be incredibly rare.  I'm going to reflect on whether I can improve detection of 'You' when 5.0 comes out.  It may require a new memloc.
